opensearch-service-domain-cdk/lib/stack-composer.ts

```diff
@@ -0,0 +1,181 @@
+            dedicatedManagerNodeCount: dedicatedManagerNodeCount,
+            warmInstanceType: warmNodeType,
+            warmNodes: warmNodeCount,
+            zoneAwarenessEnabled: zoneAwarenessEnabled,
+            zoneAwarenessAvailabilityZoneCount: zoneAwarenessAvailabilityZoneCount,
+            useUnsignedBasicAuth: useUnsignedBasicAuth,
+            fineGrainedManagerUserARN: fineGrainedManagerUserARN,
+            fineGrainedManagerUserName: fineGrainedManagerUserName,
+            fineGrainedManagerUserSecretManagerKeyARN: fineGrainedManagerUserSecretManagerKeyARN,
+            cognitoIdentityPoolId: cognitoIdentityPoolId,
+            cognitoRoleARN: cognitoRoleARN,
+            cognitoUserPoolId: cognitoUserPoolId,
+            customEndpointDomainName: customEndpointDomainName,
+            customEndpointCertificateARN: customEndpointCertificateARN,
+            customEndpointHostedZoneId: customEndpointHostedZoneId,
+            enforceHTTPS: enforceHTTPS,
+            tlsSecurityPolicy: tlsSecurityPolicy,
+            nodeToNodeEncryptionEnabled: noneToNodeEncryptionEnabled,
+            encryptionAtRestEnabled: encryptionAtRestEnabled,
+            encryptionAtRestKmsKeyARN: encryptionAtRestKmsKeyARN,
+            ebsEnabled: ebsEnabled,
+            ebsIops: ebsIops,
+            ebsVolumeSize: ebsVolumeSize,
+            ebsVolumeType: ebsVolumeType,
+            appLogEnabled: loggingAppLogEnabled,
+            appLogGroup: loggingAppLogGroupARN,
+            auditLogEnabled: loggingAuditLogEnabled,
+            auditLogGroup: loggingAuditLogGroupARN,
+            slowIndexLogEnabled: loggingSlowIndexLogEnabled,
+            slowIndexLogGroup: loggingSlowIndexLogGroupARN,
+            slowSearchLogEnabled: loggingSlowSearchLogEnabled,
+            slowSearchLogGroup: loggingSlowSearchLogGroupARN,
+            snapshotAutomatedStartHour: snapshotAutomatedStartHour,
+            vpcId: vpcId,
+            vpcSecurityGroupIds: vpcSecurityGroupIds,
+            vpcSubnetIds: vpcSubnetIds,
+            enableVersionUpgrade: enableVersionUpgrade,
+            domainRemovalPolicy: domainRemovalPolicy,
+            ...props,
+        });
+
+        function getContextForType(optionName: string, expectedType: string): any {
+            const option = scope.node.tryGetContext(optionName)
+            // Falsy check modified to not include values we want to retain
+            if (option !== false && option !== 0 && !option) {
+                return undefined
+            }
+            // Values provided by the CLI will always be represented as a string and need to be parsed
+            if (typeof option === 'string') {
+                if (expectedType === 'number') {
+                    return parseInt(option)
+                }
+                if (expectedType === 'boolean' || expectedType === 'object') {
+                    return JSON.parse(option)
+                }
```

Comment:
1. A limitation of JS, there doesn't seem to be a boolean method to parse a string and many sites recommend using JSON.parse() 🤢 
2. I believe this `object` is essentially a JSON object. I am not aware of say a JSON specific class name that we could attribute. Lmk if you have something, as this was my initial hunch but was unsuccessful in finding anything
3. The formats I expect for objects here are JSON object format `{"Version": "2012-10-17", ...}` and a string array format `["subnet-123", "subnet-124"]`. This will be clearer when I add sample types to the readme

Reply:
1. Bleh, OK.  
2. As I read it, we're providing the "type labeling" here.  It's probably just a nitpick, but I'd love to make it clear this is a JSON-ish thing.  Maybe a good compromise is to change the label to `json_object`?
3. I see.  I wouldn't invest too heavily in this sort of thing, as users that want to customize their stacks to any degree other than the most basic will probably just do it via writing TypeScript CDK.  And, frankly, I think that's what we should encourage them to do rather than providing a way to pass through complex object configuration via JSON.

Action Item - can you call out in the README that users are encouraged to customize the deployment by changing the CDK TypeScript, and that the configuration-by-context option is primarily provided for testing/development purposes?